examples/albef/model.py

```diff
@@ -431,6 +430,186 @@ def forward(
             )
 
 
+class ALBEFModelForRetrieval(nn.Module):
+    """
+    ALBEF Model for Retrieval finetuning and inference.
+
```

Comment:
Can you add more detail in the docstring explaining the different forward modes here (train vs eval, image vs text vs multimodal)